docs/security-best-practices.md

```diff
@@ -153,6 +153,108 @@ To change the default endpoint to be `localhost`-bound in all components, enable
 
 If `localhost` resolves to a different IP due to your DNS settings then explicitly use the loopback IP instead: `127.0.0.1` for IPv4 or `::1` for IPv6. In IPv6 setups, ensure your system supports both IPv4 and IPv6 loopback addresses to avoid issues.
 
+Using `localhost` may not work in environments like Docker, Kubernetes, and other environments that have non-standard networking setups.  We've documented a few working example setups for the OTLP receiver gRPC endpoint below, but other receivers and other Collector components may need similar configuration.
+
+#### Docker
+You can run the Collector in Docker using a configuration like the following:
+
+OTEL Collector config file:
+```yaml
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: my-hostname:4317 # the same hostname from your docker run command
+```
+Docker run command:
+`docker run --hostname my-hostname --name container-name -p 127.0.0.1:4567:4317 -v $(pwd):/etc/otelcol otel/opentelemetry-collector:0.104.0 --config /etc/otelcol/config.yaml`
+
+You could access it from outside that Docker network (for example on a regular program running on the host) by connecting to `127.0.0.1:4567`.
+
+#### Docker Compose
+```yaml
+services:
+  otel-collector:
```

Comment:
We are not loading the configuration for this service, while we do explicitly tell you how to load the configuration on the Docker example. Can we add that here too?

Reply:
Honestly my first instinct is to delete the configuration example from Docker and Kubernetes more than this.  We should not be trying to document how to load the config.  How does that sound?

Reply:
That's okay, but we should mention the `--hostname <hostname>` bit in the plain Docker example